helper/src/main/java/com/hazelcast/examples/helper/splitbrain/SplitBrainTestSupport.java

```diff
@@ -0,0 +1,402 @@
+                .setProperty(GroupProperty.GENERIC_OPERATION_THREAD_COUNT.getName(), "2")
+                .setProperty(GroupProperty.EVENT_THREAD_COUNT.getName(), "1")
+                .setProperty(GroupProperty.MERGE_FIRST_RUN_DELAY_SECONDS.getName(), "5")
+                .setProperty(GroupProperty.MERGE_NEXT_RUN_DELAY_SECONDS.getName(), "5");
+    }
+
+    /**
+     * Override this method to create the split-brain situation multiple-times.
+     * <p>
+     * The test will use the same members for all iterations.
+     *
+     * @return the default number of iterations
+     */
+    protected int iterations() {
+        return DEFAULT_ITERATION_COUNT;
+    }
+
+    /**
+     * Called when a cluster is fully formed. You can use this method for test initialization, data load, etc.
+     *
+     * @param instances all Hazelcast instances in your cluster
+     */
+    @SuppressWarnings("unused")
+    protected void onBeforeSplitBrainCreated(HazelcastInstance[] instances) throws Exception {
+    }
+
+    /**
+     * Called just after a split brain situation was created
+     */
+    @SuppressWarnings("unused")
+    protected void onAfterSplitBrainCreated(HazelcastInstance[] firstBrain, HazelcastInstance[] secondBrain) throws Exception {
+    }
+
+    /**
+     * Called just after the original cluster was healed again. This is likely the place for various asserts.
+     *
+     * @param instances all Hazelcast instances in your cluster
+     */
+    @SuppressWarnings("unused")
+    protected void onAfterSplitBrainHealed(HazelcastInstance[] instances) throws Exception {
+    }
+
+    /**
+     * Indicates whether test should fail when cluster does not include all original members after communications are unblocked.
+     * <p>
+     * Override this method when it is expected that after communications are unblocked some members will not rejoin the cluster.
+     * When overriding this method, it may be desirable to add some wait time to allow the split brain handler to execute.
+     *
+     * @return {@code true} if the test should fail when not all original members rejoin after split brain is
+     * healed, otherwise {@code false}.
+     */
+    protected boolean shouldAssertAllNodesRejoined() {
+        return true;
+    }
+
+    private void doIteration() throws Exception {
+        onBeforeSplitBrainCreated(instances);
+
+        createSplitBrain();
+        Brains brains = getBrains();
+        onAfterSplitBrainCreated(brains.getFirstHalf(), brains.getSecondHalf());
+
+        healSplitBrain();
+        onAfterSplitBrainHealed(instances);
+    }
+
+    protected HazelcastInstance[] startInitialCluster(Config config, int clusterSize) {
+        HazelcastInstance[] hazelcastInstances = new HazelcastInstance[clusterSize];
+        for (int i = 0; i < clusterSize; i++) {
+            hazelcastInstances[i] = HazelcastInstanceFactory.newHazelcastInstance(config, createInstanceName(config),
```

Comment:
This creates a `HazelcastInstance` with a `FirewallingConnectionManager`, so we can block and unblock the communication between nodes.